packages/controller/lib/setup/setupBackup.js

```diff
@@ -55,55 +55,45 @@ class BackupRestore {
 
-        this.reloadAdapterObject = this.reloadAdapterObject.bind(this);
-        this._setStateHelper = this._setStateHelper.bind(this);
-        this._setObjHelper = this._setObjHelper.bind(this);
-        this.reloadAdaptersObjects = this.reloadAdaptersObjects.bind(this);
     } // endConstructor
 
     // --------------------------------------- BACKUP ---------------------------------------------------
-    _copyFile(id, srcPath, destPath, callback) {
-        this.objects.readFile(id, srcPath, '', (err, data) => {
-            try {
-                data && fs.writeFileSync(destPath, data);
-            } catch (e) {
-                console.log('Can not copy File ' + id + '/' + srcPath + ' to ' + destPath + ': ' + e);
+    async _copyFile(id, srcPath, destPath) {
+        try {
+            const data = await this.objects.readFileAsync(id, srcPath, '');
+            if (data) {
+                if (data.data !== undefined) {
+                    fs.writeFileSync(destPath, data.data);
+                } else {
+                    fs.writeFileSync(destPath, data);
+                }
             }
-            return tools.maybeCallback(callback);
-        });
+        } catch (err) {
+            console.log(`Can not copy File ${id}${srcPath} to ${destPath}: ${err.message || err}`);
+        }
     }
 
-    copyDir(id, srcPath, destPath, callback) {
-        let count = 0;
+    async copyDir(id, srcPath, destPath) {
         !fs.existsSync(destPath) && fs.mkdirSync(destPath);
 
-        this.objects.readDir(id, srcPath, (err, res) => {
+        try {
+            const res = await this.objects.readDirAsync(id, srcPath);
             if (res) {
                 for (let t = 0; t < res.length; t++) {
                     if (res[t].isDir) {
-                        count++;
-                        this.copyDir(id, `${srcPath}/${res[t].file}`, `${destPath}/${res[t].file}`, () => {
-                            if (!--count) {
-                                return tools.maybeCallback(callback);
-                            }
-                        });
+                        await this.copyDir(id, `${srcPath}/${res[t].file}`, `${destPath}/${res[t].file}`);
                     } else {
                         !fs.existsSync(destPath) && fs.mkdirSync(destPath);
 
-                        count++;
-
-                        this._copyFile(id, `${srcPath}/${res[t].file}`, `${destPath}/${res[t].file}`, () => {
-                            if (!--count) {
-                                return tools.maybeCallback(callback);
-                            }
-                        });
+                        await this._copyFile(id, `${srcPath}/${res[t].file}`, `${destPath}/${res[t].file}`);
                     }
                 }
             }
-            if (!count) {
-                return tools.maybeCallback(callback);
+        } catch (err) {
+            if (!err.toString().includes('Not exists')) {
```

Comment:
err alsowhen "not exists" should only be Error object from my rememberings ...

Reply:
err.message used